### PR TITLE
Update Hypertune example to use the new Feature Flags pattern

### PR DESCRIPTION
**Pull request by miraan**
Update Hypertune example to use the new Feature Flags pattern

### CODE REVIEW

File: edge-middleware/feature-flag-hypertune/.env [hunk 1]
```diff
@@ -0,0 +1,4 @@
+HYPERTUNE_FRAMEWORK=nextApp
+HYPERTUNE_PLATFORM=vercel
+HYPERTUNE_OUTPUT_DIRECTORY_PATH=generated
+HYPERTUNE_GET_HYPERTUNE_IMPORT_PATH=../lib/getHypertune
```
Comment (dferber90) [inline edge-middleware/feature-flag-hypertune/.env:4]: Is HYPERTUNE_ADMIN_TOKEN missing here?

File: edge-middleware/feature-flag-hypertune/app/.well-known/vercel/flags/route.ts
Comment (dferber90) [inline edge-middleware/feature-flag-hypertune/app/.well-known/vercel/flags/route.ts:1]: It looks like something is forcing the browser to refresh. And that seems to be mounting/unmounting the FlagValues, which then in turn triggers network request(s) to decrypt the flag values detected in the DOM.

This looks like a bug

https://github.com/vercel/examples/assets/1765075/0b075788-be51-450b-bb5c-64916d6aff70


_Not related to this file, but using this so we can have a threaded convesation_

File: edge-middleware/feature-flag-hypertune/app/page.tsx [hunk 2]
```diff
@@ -26,13 +27,16 @@ export default async function Home() {
           with Vercel Edge Config to initialize the Hypertune SDK with near-zero
           latency on the server so you can access your feature flags and run A/B
           tests with no performance impact to your app. It also shows how to
-          integrate with the Vercel Toolbar to let you easily set local flag
-          overrides while developing.
+          integrate with the Vercel Toolbar to let you easily view and override
+          your feature flags without leaving your frontend. Finally, it shows
+          how to use the Vercel Feature Flags pattern to use flags on a static
```
Comment (dferber90) [inline edge-middleware/feature-flag-hypertune/app/page.tsx:32]: To clarify a bit: we consider the ability to define flags in code as our feature flags pattern, and the ability to precompute some flags and use middleware to route between static pages is basically just an extension of that.

**Review by dferber90 — CHANGES_REQUESTED** [review]
Looks super promising!

File: edge-middleware/feature-flag-hypertune/.env (hunk 1 above)
Comment (miraan) [reply edge-middleware/feature-flag-hypertune/.env:4]: The following environment variables are set via the deploy flow: NEXT_PUBLIC_HYPERTUNE_TOKEN, HYPERTUNE_ADMIN_TOKEN, EDGE_CONFIG, EDGE_CONFIG_HYPERTUNE_ITEM_KEY, FLAGS_SECRET

They are needed in production. The instructions say to pull these to .env.development.local to run the project locally.

The environment variables here in .env are not needed in production. They are used by the CLI tool for codegen only.

File: edge-middleware/feature-flag-hypertune/app/page.tsx (hunk 2 above)
Comment (miraan) [reply edge-middleware/feature-flag-hypertune/app/page.tsx:32]: We show the middleware + routing between static pages in this example so thought I'd call it out. But will update the wording here!

**Review by dferber90 — APPROVED** [review]

**Pull request merged**